### PR TITLE
Dremio datasource

### LINKED ISSUE #1685: Add Dremio as a supported datasource :electric_plug:  :mag:
We need to increase the number of datasources that MindsDB supports. This task should add a new datasource for connecting to Dremio. You can get more info on [dremio](https://docs.dremio.com/client-applications/python/) docs. 
If possible, please include a test for the datasource. You can check the example [here](https://github.com/mindsdb/datasources/tree/staging/tests).

>Note: if you are familiar with another datasource that MindsDB doesn't support, feel free to work on that and make a PR.

## Steps :male_detective: :female_detective: 

- [ ] Frok datasources repo
- [ ] Add new datasource file in https://github.com/mindsdb/datasources/tree/staging/mindsdb_datasources/datasources
- [ ] Add test(optional)
- [ ] Make a PR to staging branch

## Additional rewards :1st_place_medal: 

Each datasource PR brings  :four:  point for entry into the draw for a :computer:  Deep Learning Laptop powered by the NVIDIA RTX 3080 Max-Q GPU or other swag :shirt: :bear: .  For more info check out https://mindsdb.com/hacktoberfest/

### PR DESCRIPTION
**Pull request by ktyborowski**
Fixes mindsdb/mindsdb#1685

Initial implementation of the Dremio datasource.

What has been done:

1. Created and initial implementation of Dremio datasource based on the existing snowflake datasource and Dremio docs.
2. Added a very simple test boilerplate. This requires a Dremio instance with a database so for now I left the rest for discussion.

### CODE REVIEW

File: mindsdb_datasources/datasources/dremio_ds.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import pandas as pd
+import pyodbc
```
Comment (ZoranPandovski) [inline mindsdb_datasources/datasources/dremio_ds.py:2]: Can you please add this to the requirements?

File: mindsdb_datasources/datasources/dremio_ds.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+from mindsdb_datasources.datasources.data_source import SQLDataSource
+
+
+class DremioDS(SQLDataSource):
```
Comment (ZoranPandovski) [inline mindsdb_datasources/datasources/dremio_ds.py:6]: Please add this to __init__.py

File: mindsdb_datasources/datasources/dremio_ds.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    def __init__(self, query, host, user, password, 
+                 driver='Dremio ODBC Driver 64-bit', port=31010):
```
Comment (ZoranPandovski) [inline mindsdb_datasources/datasources/dremio_ds.py:8]: Maybe you can add warning in the __init__ py same as we did for snowflake, that driver is required

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb_datasources/datasources/dremio_ds.py (hunk 1 above)
Comment (ktyborowski) [reply mindsdb_datasources/datasources/dremio_ds.py:2]: I added information about the requirement to the init notification as that's how it's done in the snowflake_ds. If you would prefer this to be in the requirements let me know.

File: mindsdb_datasources/datasources/dremio_ds.py (hunk 2 above)
Comment (ktyborowski) [reply mindsdb_datasources/datasources/dremio_ds.py:6]: Added.

File: mindsdb_datasources/datasources/dremio_ds.py (hunk 3 above)
Comment (ktyborowski) [reply mindsdb_datasources/datasources/dremio_ds.py:8]: Done, also added pyodbc requirement to the notification as it's done in the snowflake_ds.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**